stock_storage_type/models/stock_location_storage_type.py

```diff
@@ -114,3 +111,39 @@ def _compute_has_restrictions(self):
                     slst.max_weight,
                 ]
             )
+
+    def _domain_location_storage_type(self, candidate_locations, quants, products):
+        """compute a domain which applies the constraint of the
+        stock.location.storage.type to select locations among candidate
+        locations.
+        """
+        self.ensure_one()
+        location_domain = [
+            ('id', 'in', candidate_locations.ids),
+            ('allowed_location_storage_type_ids', '=', self.id),
+        ]
+        if self.only_empty:
```

Comment:
From here, it is not related to storage type. Any chance to extract this in another method that is not recomputed for each storage type?

Reply:
It's related to the configuration of the storage type (only_empty, do_not_mix_products, do_not_mix_lots).
We could optimize this by grouping the storage types by similar configuration to reduce the number of queries, but it isn't required now. Adding a todo